### ISSUE TITLE
Use bazel repository_rule to configure clang and go, rather than setup_bazel.sh

Note: This requires bazelhead! [phab:D756]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1039]
_Created by **shahms** at 2016-03-04 23:36:59_: 

Automatically configure go/clang

**Comment by kythe** [comment]
_**zarko** wrote at 2016-03-04 23:55:41_:

This fails to build on OS X:
```
ERROR: /Users/zarko/kythe-ci/third_party/go/BUILD:26:1: error executing shell command: 'set -e
export GOROOT=$PWD/external/local_goroot
export CC=external/local_config_cc/osx_gcc_wrapper.sh
export CGO_CFLAGS="-DLEVELDB_PLATFORM_POSIX -DOS_MACOSX -isystem $PWD/third_party/leveldb/inclu...' failed: bash failed: error executing command
  (cd /private/var/tmp/_bazel_zarko/2be6994bf77f75bac8f1433eb6e9dc33/kythe-ci && \
  exec env - \
    PATH=/Users/zarko/bin:/usr/local/git/current/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/go/bin:/Library/TeX/texbin:/Users/zarko/bazel/output:/Users/zarko/arcanist/arcanist/bin \
    TMPDIR=/var/folders/ft/bv0xnfpx2bqbqc1zp24gc4n8001p27/T/ \
  /bin/bash -c 'set -e
export GOROOT=$PWD/external/local_goroot
export CC=external/local_config_cc/osx_gcc_wrapper.sh
export CGO_CFLAGS="-DLEVELDB_PLATFORM_POSIX -DOS_MACOSX -isystem $PWD/third_party/leveldb/include -isystem $PWD/bazel-out/local_darwin-fastbuild/genfiles/third_party/leveldb/include -isystem $PWD/third_party/snappy -isystem $PWD/bazel-out/local_darwin-fastbuild/genfiles/third_party/snappy -isystem $PWD/external/bazel_tools/tools/cpp/gcc3 -iquote $PWD/. -iquote $PWD/bazel-out/local_darwin-fastbuild/genfiles -iquote $PWD/external/bazel_tools -iquote $PWD/bazel-out/local_darwin-fastbuild/genfiles/external/bazel_tools"
export CGO_LDFLAGS="-lstdc++ $PWD/bazel-out/local_darwin-fastbuild/bin/third_party/leveldb/libleveldb.a $PWD/bazel-out/local_darwin-fastbuild/bin/third_party/snappy/libsnappy.a"
export GOPATH="$PWD/third_party/go"
external/local_goroot/bin/go build -a  -o bazel-out/local_darwin-fastbuild/bin/third_party/go/levigo.a github.com/jmhodges/levigo'): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 2.
# runtime/cgo
fork/exec external/local_config_cc/osx_gcc_wrapper.sh: no such file or directory
```

**Comment by kythe** [comment]
_**shahms** wrote at 2016-03-05 00:31:57_:

  - Include PWD for relative CC paths to work on OSX

# Updating (earlier comment): Use bazel repository_rule to configure clang and go, rather than setup_bazel.sh

Note: This requires bazelhead!

**Comment by kythe** [comment]
_**zarko** wrote at 2016-03-05 00:40:02_:

The last update now fixes the OS X problem.

**Comment by kythe** [comment]
_**zarko** wrote at 2016-03-05 01:01:52_:

The LLVM sub-build now fails:

`checking for C compiler default output file name... configure: error: C compiler cannot create executables`

It looks like the paths in bazel-genfiles/tools/modules/compiler_info might be bad:

```
BAZEL_CC=external/local_config_cc/osx_gcc_wrapper.sh
BAZEL_C_COMPILER=osx_gcc_wrapper.sh
```

(these are absolute paths to Clang on Linux).

`configure` is invoked as

`../configure CC=osx_gcc_wrapper.sh CXX=external/local_config_cc/osx_gcc_wrapper.sh++`

**Comment by kythe** [comment]
_**zarko** wrote at 2016-03-05 01:13:15_:

On OS X, the previous paths in compiler_info.txt were:
```
BAZEL_CC=/Users/zarko/kythe-ci/tools/cpp/clang
BAZEL_C_COMPILER=/Users/zarko/kythe-ci/tools/cpp/clang
```
These files are scripts set up by `setup_bazel.sh`.

**Comment by kythe** [comment]
_**shahms** wrote at 2016-03-05 01:40:01_:

Interesting, thanks Luke.  The "clang" script that's setup is in fact the same as the osx_gcc_wrapper.sh, just with a different name.  I'll dig in a bit more.

**Comment by kythe** [comment]
_**shahms** wrote at 2016-03-05 08:12:42_:

  - Fixes for relative paths to wrappers on OSX

# Updating (earlier comment): Use bazel repository_rule to configure clang and go, rather than setup_bazel.sh

Note: This requires bazelhead!

**Comment by kythe** [comment]
_**zarko** wrote at 2016-03-07 04:54:49_:

Works on OS X now. Thanks!

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2016-03-07 04:54:49_

**Comment by kythe** [comment]
_**shahms** closed via commit 113eb559d0e81e47900e630a1cd40df07ed32346 at 2016-03-07 16:29:33_
